src/gleam/float.gleam

```diff
@@ -572,3 +572,83 @@ pub fn multiply(a: Float, b: Float) -> Float {
 pub fn subtract(a: Float, b: Float) -> Float {
   a -. b
 }
+
+/// Returns the natural logarithm (base e) of the given as a `Result`. If the
+/// input is less than or equal to 0, returns `Error(Nil)`.
+///
+/// ## Examples
+///
+/// ```gleam
+/// log(1.0)
+/// // -> Ok(0.0)
+/// ```
+///
+/// ```gleam
+/// log(2.718281828459045)  // e
+/// // -> Ok(1.0)
+/// ```
+///
+/// ```gleam
+/// log(10.0)
+/// // -> Ok(2.302585092994046)
+/// ```
+///
+/// ```gleam
+/// 100.0 |> log
+/// // -> Ok(4.605170185988092)
+/// ```
+///
+/// ```gleam
+/// log(0.0)
+/// // -> Error(Nil)
+/// ```
+///
+/// ```gleam
+/// log(-1.0)
+/// // -> Error(Nil)
+/// ```
+///
+pub fn log(x: Float) -> Result(Float, Nil) {
+  // In the following check:
+  // 1. If x is negative then return an error as the natural logarithm
+  //    of a negative number is undefined (would be a complex number)
+  // 2. If x is 0 then return an error as the natural logarithm of 0
+  //    approaches negative infinity
+  case x <=. 0.0 {
+    True -> Error(Nil)
+    False -> Ok(do_log(x))
+  }
+}
+
+@external(erlang, "math", "log")
+@external(javascript, "../gleam_stdlib.mjs", "log")
+fn do_log(x: Float) -> Float
+
+/// Returns e (Euler's number) raised to the power of the given exponent, as
+/// a `Float`.
+///
+/// ## Examples
+///
+/// ```gleam
+/// exp(0.0)
+/// // -> Ok(1.0)
+/// ```
+///
+/// ```gleam
+/// exp(1.0)
+/// // -> Ok(2.718281828459045)
+/// ```
+///
+/// ```gleam
+/// exp(-1.0)
+/// // -> Ok(0.36787944117144233)
+/// ```
+///
+/// ```gleam
+/// 2.0 |> exp
+/// // -> Ok(7.38905609893065)
+/// ```
+///
+@external(erlang, "math", "exp")
+@external(javascript, "../gleam_stdlib.mjs", "exp")
+pub fn exp(x: Float) -> Float
```

Comment:
No abbreviations please

Reply:
I'm not sure what you mean? Do you want me to call it `natural_logarithm` and `exponential`?

Reply:
We never use abbreviations, so aye whatever these abbreviations are short for.

Reply:
okay, I updated it to `logarithm` and `exponential`!